### ISSUE TITLE
Segfault when running model in eval mode and using .numpy()

### ISSUE THREAD

**Issue by kierkegaard13** [pytorch/pytorch#9899]
If you have a question or would like help and support, please ask at our
[forums](https://discuss.pytorch.org/).

If you are submitting a feature request, please preface the title with [feature request].
If you are submitting a bug report, please fill in the following details.

## Issue description
I'm training a model using pytorch and when I try and create predictions using the same model and export the predictions to numpy before returning from my predict function, I get a segfault. This only occurs when running on a cpu (no .cuda). Further, there are 3 solutions that I've identified. I've reproduced this using a cleaned version of the Beijing pollution dataset from the UCI repository:
https://archive.ics.uci.edu/ml/datasets/Beijing+PM2.5+Data#
1. Run on a gpu using .cuda
2. Use .numpy after returning predictions from predict function
3. Wrap predict call in `with torch.no_grad()`

## Code example

Please try to provide a minimal example to repro the bug.
Error messages and stack traces are also helpful.

I've included a simple python script with the cleaned beijing data and target csv's.

[btest.zip](https://github.com/pytorch/pytorch/files/2233402/btest.zip)

## System Info
Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

- PyTorch or Caffe2: PyTorch
- How you installed PyTorch (conda, pip, source): conda
- Build command you used (if compiling from source): 
- OS: Ubuntu 16.04
- PyTorch version: 0.4.0
- Python version: 3.6.5
- CUDA/cuDNN version: 9.1/7.1
- GPU models and configuration: 1 GTX 1080
- GCC version (if compiling from source): 
- CMake version: 
- Versions of any other relevant libraries:

**Comment by mruberry** [comment]
Closing due to age. If this issue is still occurring please file a new issue with a more recent version of PyTorch.
